### PR TITLE
Fix ProtoBuf

### PR DESCRIPTION
**Pull request by chrisblatchley**
sorry to say, the bug was pretty small. :)

in the module encode, we were only increasing the buffer written length by 1 byte per module, which i thought was weird but maybe it was compressed.

nope: it was only writing the tag, not the actual data. main change here is the following:

```c
        bool status = pb_encode_submessage(stream, TeslaBMS_Pack_Module_fields, &source->modarr[i]);
        if (status) {
            printf("\n...%i bytes written so far...\n", stream->bytes_written);
        }
```

the `pb_encode_submessage` was missing ;) it was just a weird looking if statement.

check out the rest of the changes too. i changed the decode to include a `while(ostream->bytes_left)` but i think you can get rid of that if you want? i don't know i haven't tried it i'm going to bed. :)

### CODE REVIEW

File: src/TeslaBMS.cpp [hunk 1]
```diff
@@ -67,133 +62,106 @@ void module_array_maker(ModuleList *list){
...
         {
             const char * error = PB_GET_ERROR(stream);
             printf("encode_modules error: %s", error);
             return false;
         }
-        if (stream, TeslaBMS_Pack_Module_fields, &source->modarr[i])
-        {
-            printf("\n %i \n",stream-> bytes_written);
+        
+        bool status = pb_encode_submessage(stream, TeslaBMS_Pack_Module_fields, &source->modarr[i]);
```
Comment (chrisblatchley) [inline src/TeslaBMS.cpp:78]: crux of the bug is this line here

File: src/TeslaBMS.cpp [hunk 2]
```diff
@@ -67,133 +62,106 @@ void module_array_maker(ModuleList *list){
...
 
-    TeslaBMS_Pack_Module module;
-    if(!pb_decode(istream, TeslaBMS_Pack_Module_fields, &module)){
-        const char * error = PB_GET_ERROR(istream);
-        printf("module_decode error: %s", error);
-        return false;
-    }
-    
-    modulelist_add_module(dest, module);
+    while(istream->bytes_left) {
```
Comment (chrisblatchley) [inline src/TeslaBMS.cpp]: TODO: see if it works without this part.

Comment (chrisblatchley) [reply src/TeslaBMS.cpp]: DONE: i tried it anyway and yup it's redundant so i took it out.

File: src/TeslaBMS.cpp (hunk 1 above)
Comment (ahrenswett) [reply src/TeslaBMS.cpp:78]: 🤦‍♂️ughhhh good catch, looks like I almost had it :)

**Pull request merged**